src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+      cam1.eyeY - cam1.centerY,
+      cam1.eyeZ - cam1.centerZ
+    );
+    // Then linearly interpolate them by amt.
+    const lerpedDist = (1 - amt) * dist0 + amt * dist1;
+
+    // Next, calculate the ratio to interpolate the eye and center by a constant
+    // ratio for each camera. This ratio is the same for both. Also, with this ratio
+    // of points, the distance is the minimum distance of the two points of
+    // the same ratio.
+    // With this method, if the viewpoint is fixed, linear interpolation is performed
+    // at the viewpoint, and if the center is fixed, linear interpolation is performed
+    // at the center, resulting in reasonable interpolation. If both move, the point
+    // halfway between them is taken.
+    const eyeDiff = new p5.Vector(
+      cam0.eyeX - cam1.eyeX, cam0.eyeY - cam1.eyeY, cam0.eyeZ - cam1.eyeZ
+    );
+    const diffDiff = new p5.Vector(
+      cam0.eyeX - cam1.eyeX - cam0.centerX + cam1.centerX,
+      cam0.eyeY - cam1.eyeY - cam0.centerY + cam1.centerY,
+      cam0.eyeZ - cam1.eyeZ - cam0.centerZ + cam1.centerZ
+    );
+    // Suppose there are two line segments. Consider the distance between the points
+    // above them as if they were taken in the same ratio. This calculation figures out
+    // a ratio that minimizes this.
+    // Each line segment is, a line segment connecting the viewpoint and the center
+    // for each camera.
+    const divider = diffDiff.magSq();
+    let ratio = 1; // default.
+    if (divider > 0.000001){
+      ratio = p5.Vector.dot(eyeDiff, diffDiff) / divider;
+      ratio = Math.max(0, Math.min(ratio, 1));
+    }
+
+    // Take the appropriate proportions and work out the points
+    // that are between the new viewpoint and the new center position.
+    const medium0X = (1-ratio) * cam0.eyeX + ratio * cam0.centerX;
+    const medium0Y = (1-ratio) * cam0.eyeY + ratio * cam0.centerY;
+    const medium0Z = (1-ratio) * cam0.eyeZ + ratio * cam0.centerZ;
+
+    const medium1X = (1-ratio) * cam1.eyeX + ratio * cam1.centerX;
+    const medium1Y = (1-ratio) * cam1.eyeY + ratio * cam1.centerY;
+    const medium1Z = (1-ratio) * cam1.eyeZ + ratio * cam1.centerZ;
+
+    const lerpedMediumX = medium0X * (1-amt) + medium1X * amt;
+    const lerpedMediumY = medium0Y * (1-amt) + medium1Y * amt;
+    const lerpedMediumZ = medium0Z * (1-amt) + medium1Z * amt;
+
+    // Prepare each of rotation matrix from their camera matrix
+    const mat0 = cam0.cameraMatrix.mat4;
+    const mat1 = cam1.cameraMatrix.mat4;
+    const m0 = [
+      mat0[0], mat0[1], mat0[2],
+      mat0[4], mat0[5], mat0[6],
+      mat0[8], mat0[9], mat0[10]
+    ];
+    const m1 = [
+      mat1[0], mat1[1], mat1[2],
+      mat1[4], mat1[5], mat1[6],
+      mat1[8], mat1[9], mat1[10]
+    ];
+    // Create the inverse matrix of mat0 by transposing mat0,
+    // and multiply it to mat1 from the right.
+    // This matrix represents the difference between the two.
+    const m = [
```

Comment:
Maybe we could make this out of `p5.Matrix` to make the transpose + multiply clearer? It looks like we support mat3s there if you do `new p5.Matrix('mat3', [...])`. I'm not certain all the operations work on mat3s, so if they don't, this is also fine